packages/analytics-core/src/core-client.ts

```diff
@@ -98,37 +80,57 @@ export class AmplitudeCore<T extends Config> implements CoreClient<T> {
 
   async add(plugin: Plugin) {
     if (!this.config) {
-      this.pluginQueue.push(this.add.bind(this, plugin));
+      this.q.push(this.add.bind(this, plugin));
       return;
     }
     return this.timeline.register(plugin, this.config);
   }
 
   async remove(pluginName: string) {
     if (!this.config) {
-      this.pluginQueue.push(this.remove.bind(this, pluginName));
+      this.q.push(this.remove.bind(this, pluginName));
       return;
     }
     return this.timeline.deregister(pluginName);
   }
 
-  async dispatch(event: Event) {
+  dispatchWithCallback(event: Event, callback: (result: Result) => void): void {
+    if (!this.config) {
+      return callback(buildResult(event, 0, CLIENT_NOT_INITIALIZED));
```

Comment:
If `config` is not ready (not initialized), this will resolve the promise with `CLIENT_NOT_INITIALIZED`. Are we just dropping these events? I am not sure how this handles the case when `track` is called before `runAttributionStrategy`, that the tracked event will be sent after identify. Could you elaborate more? Thanks!

Reply:
`dispatchWithCallback` is designed to only be called when in `this.delayedQ` as this ensures that `this.config` is defined. I only added this case just in case it's ever called (but probably never will) and it will return this error so dev would see.

Reply:
I also created `dispatchWithCallback` to create a way where `dispatch` can always return `Promise<Result>`. The alternative was to add to fn call to queue but return `undefined`, which I personally didn't like.

https://github.com/amplitude/Amplitude-TypeScript/blob/DXOC-244/packages/analytics-core/src/core-client.ts#L106-L108